vault/core.go

```diff
@@ -2632,6 +2647,91 @@ func (c *Core) SetConfig(conf *server.Config) {
 	c.logger.Debug("set config", "sanitized config", string(bz))
 }
 
+func (c *Core) getCustomHeadersListenerList(listenerAdd string) []*ListenerCustomHeaders {
+	customHeaders := c.customListenerHeader.Load()
+	if customHeaders == nil {
+		return nil
+	}
+
+	customHeadersList := customHeaders.([]*ListenerCustomHeaders)
+	if customHeadersList == nil {
+		return nil
+	}
+
+	// either looking for a specific listener, or if listener address isn't given,
+	// checking for all available listeners
+	var lch []*ListenerCustomHeaders
+	if listenerAdd == "" {
+		lch = customHeadersList
+	} else {
```

Comment:
Totally fine to ignore this, but: it'd be more idiomatic to do early returns instead of declaring lch, using an else, doing a break, etc.